node/src/neighborhood/node_location.rs

```diff
@@ -0,0 +1,208 @@
+        let node_location =
+            get_node_location(Some(IpAddr::V4(Ipv4Addr::new(125, 125, 125, 1)))).unwrap();
+
+        assert_eq!(node_location.country_code, "CN");
+        assert_eq!(node_location.free_world_bit, false);
+    }
+
+    #[test]
+    fn construct_node_record_metadata_with_free_world_bit() {
+        let mut metadata = NodeRecordMetadata::new();
+        metadata.node_location_opt = get_node_location(Some(IpAddr::V4(Ipv4Addr::new(1, 1, 1, 1))));
+        assert_eq!(
+            metadata.node_location_opt.as_ref().unwrap(),
+            &NodeLocation {
+                country_code: "AU".to_string(),
+                free_world_bit: true
+            }
+        );
+        assert_eq!(
+            metadata.node_location_opt.as_ref().unwrap().free_world_bit,
+            true
+        );
+        assert_eq!(
+            metadata.node_location_opt.as_ref().unwrap().country_code,
+            "AU"
+        );
+    }
+
+    #[test]
+    fn construct_node_record_for_test() {
+        let mut node_record = make_node_record(1111, true);
+        let country_record = pick_country_code_record(3333);
+        node_record.metadata.node_location_opt = Some(NodeLocation {
+            country_code: country_record.1.clone(),
+            free_world_bit: country_record.2,
+        });
+        node_record.metadata.node_addr_opt =
+            Some(NodeAddr::new(&country_record.0, &[8000 % 10000]));
+        node_record.inner.country_code = country_record.1;
+
+        assert_eq!(
+            node_record.metadata.node_location_opt,
+            Some(NodeLocation {
+                country_code: "AU".to_string(),
+                free_world_bit: true
+            })
+        )
+    }
+
+    #[test]
+    fn create_gossip_node_with_addr_and_country_code_reveal_results_in_node_with_addr_adn_free_world_bit(
+    ) {
+        let mut node = make_node_record(2222, true);
+        let country_record = pick_country_code_record(2222);
+        println!("node: {:?}", node);
+        node.metadata.node_location_opt = Some(NodeLocation {
+            country_code: country_record.1.clone(),
+            free_world_bit: country_record.2,
+        });
+        node.metadata.node_addr_opt = Some(NodeAddr::new(&country_record.0, &[8000 % 10000]));
+        node.inner.country_code = country_record.1;
+        let db = db_from_node(&node);
+        let builder = GossipBuilder::new(&db);
+
+        let builder = builder.node(node.public_key(), true);
+
+        let mut gossip = builder.build();
+        let gossip_result = gossip.node_records.remove(0);
+        let node_record = NodeRecord::try_from(&gossip_result).unwrap();
+
+        println!("node_record: {:?}", node_record);
+        assert_eq!(
+            gossip_result.node_addr_opt.unwrap(),
+            node.node_addr_opt().unwrap()
+        );
+        assert_eq!(node_record.inner.country_code, "US")
+    }
+}
+
+#[allow(dead_code, unused_imports)]
+#[cfg(not(test))]
```

Comment:
Hmmm. These **should** be simply 

```Rust
#[cfg(test)]
```

but there is a bigger problem than that...

...because these test don't apply for "unit tests". And I'm resisting to pull them through some adjustments that I believe they need and then to call them "integration tests". I believe these either should be co called "ocassional tests" which is a special category that we have discussed multiple times, but was never implemented to this date. (Because it requires some thinking about our test tree in a bigger picture.) or it's an absolutely legitimate answer, I think, saying we don't need these anymore, as they can be understood more like spikes that gave a rough idea about the time requirements when we were wondering exactly about this, we might do that less in the future, because we now know the concept is bearable. 

#### Therefore I'm suggesting you to bring this comment up in front of Dan and ask him of his decisive opinion.